src/eddymotion/utils.py

```diff
@@ -0,0 +1,149 @@
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Utils to sort the DWI data volume indices"""
+
+import numpy as np
+
+
+def linear_action(size=None, **kwargs):
+    """
+    Sort the DWI data volume indices linearly
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+
+    Examples
+    --------
+    >>> list(linear_action(10))
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+    """
+    if size is None and 'bvals' in kwargs:
+        size = len(kwargs['bvals'])
+    if size is None:
+        raise TypeError("Cannot build iterator without size")
+
+    return range(size)
+
+
+def random_action(size=None, **kwargs):
+    """Sort the DWI data volume indices.
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+    """
+
+    if size is None and 'bvals' in kwargs:
+        size = len(kwargs['bvals'])
+    if size is None:
+        raise TypeError("Cannot build iterator without size")
+
+    _seed = kwargs.get('seed', None)
+    if kwargs.get('seed', None) or kwargs.get('seed', None) == 0:
+        _seed = 20210324 if kwargs.get('seed', None) is True else kwargs.get('seed', None)
+
+    rng = np.random.default_rng(_seed)
+
+    index_order = np.arange(size)
+    rng.shuffle(index_order)
+
+    return index_order.to_list()
+
+
+def bvalue_action(size=None, **kwargs):
+    """
+    Sort the DWI data volume indices in ascending order based on the last column of gradients.
+
+    Parameters
+    ----------
+    bvalues : :obj:`list`
+        List of b-values corresponding to all orientations of the dataset.
+
+    Examples
+    --------
+    >>> bvalue_action(bvals=[[1, 1, 1, 1], [4, 4, 4, 4], [2, 2, 2, 2]])
+    [0, 2, 1]
+
+    Returns
+    -------
+    :obj:`list` of :obj:`int`
+        The sorted index order.
+    """
+    bvals = kwargs.get('bvals', None)
+    if bvals is None:
+        raise TypeError('Keyword argument bvals is required')
+    indexed_bvals = sorted([(round(sum(sublist), 2), i) for i, sublist in enumerate(bvals)])
+    return [index[1] for index in indexed_bvals]
+
+
+def centralsym_action(size=None, **kwargs):
+    """
+    Sort the DWI data volume indices in a central symmetric manner.
+
+    Parameters
+    ----------
+    size : :obj:`int`
+        Number of volumes in the dataset
+        (for instance, the number of orientations in a DWI)
+
+    Examples
+    --------
+    >>> centralsym_action(10)
+    [5, 4, 6, 3, 7, 2, 8, 1, 9, 0]
+    >>> centralsym_action(11)
+    [5, 4, 6, 3, 7, 2, 8, 1, 9, 0, 10]
```

Comment:
If we turn this into a generator, then we need to expand lists before testing:
```suggestion
    >>> list(centralsym_iterator(10))
    [5, 4, 6, 3, 7, 2, 8, 1, 9, 0]
    >>> list(centralsym_iterator(11))
    [5, 4, 6, 3, 7, 2, 8, 1, 9, 0, 10]
```